src/main.go

```diff
@@ -22,7 +22,7 @@ func NewTemporalCloudProvider(temporalCloudAdapter ITemporalCloudAdapter, config
 
 	err := temporalCloudAdapter.Flags().Parse(os.Args)
 	if err != nil {
-		return nil, fmt.Errorf("unable to parse flags: %v", err)
+		return nil, fmt.Errorf("failed to parse flags: %v", err)
```

Comment:
The error message was updated to improve clarity by changing 'unable to parse flags' to 'failed to parse flags'. This is a positive change but consider standardizing your error messages across the project for consistency, as other error messages use both 'failed to' and 'unable to'.